annotator/ffa/unknownbinariesanno/filterknownbinaries.go

```diff
@@ -0,0 +1,100 @@
+// Package unknownbinariesanno removes all packages extracted by unknown binaries
+// filters out the known binaries, and records the remaining as a finding.
+package unknownbinariesanno
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/google/osv-scalibr/annotator"
+	"github.com/google/osv-scalibr/annotator/ffa/unknownbinariesanno/internal/dpkgfilter"
+	"github.com/google/osv-scalibr/annotator/ffa/unknownbinariesanno/internal/filter"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem/ffa/unknownbinariesextr"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/plugin"
+)
+
+// Name of the plugin
+const Name = "ffa/unknownbinaries"
+
+// List of filters to apply to exclude known binaries
+var filters = []filter.Filter{
+	dpkgfilter.DpkgFilter{},
+}
+
+// Annotator further processes the UnknownBinaryExtractor
+type Annotator struct {
+}
+
+// Annotate removes all packages extracted by unknown binaries,
+// filters out the known binaries, and records the remaining as a finding.
+func (anno *Annotator) Annotate(ctx context.Context, input *annotator.ScanInput, inv *inventory.Inventory) error {
+	unknownBinariesSet := map[string]*extractor.Package{}
+	filteredPackages := make([]*extractor.Package, 0, len(inv.Packages))
+	for _, e := range inv.Packages {
+		if !slices.Contains(e.Plugins, unknownbinariesextr.Name) {
+			filteredPackages = append(filteredPackages, e)
+			continue
+		}
+
+		unknownBinariesSet[e.Locations[0]] = e
+	}
+
+	// First account for all the files we have successfully extracted.
+	for _, e := range filteredPackages {
+		for _, location := range e.Locations {
+			delete(unknownBinariesSet, location)
```

Comment:
Does Go allow removing from slices while iterating on them?

Reply:
I don't think so, but I don't think I'm deleting from a slice while iterating over it here? I build the set above, and remove it in a separate loop here.